cpp/src/io/avro/reader_impl.cu

```diff
@@ -407,8 +420,9 @@ table_with_metadata reader::impl::read(avro_reader_options const &options, cudaS
         for (const auto &sym : col_schema.symbols) { dictionary_data_size += sym.length(); }
       }
 
-      hostdevice_vector<uint8_t> global_dictionary(
-        total_dictionary_entries * sizeof(gpu::nvstrdesc_s) + dictionary_data_size);
+      hostdevice_vector<gpu::nvstrdesc_s> global_dictionary(total_dictionary_entries +
+                                                            dictionary_data_size);
+
       if (total_dictionary_entries > 0) {
         size_t dict_pos = total_dictionary_entries * sizeof(gpu::nvstrdesc_s);
         for (size_t i = 0; i < column_types.size(); ++i) {
```

Comment:
3 lines below, there's `reinterpret_cast<gpu::nvstrdesc_s *>(global_dictionary.host_ptr())` which is redundant now.